src/metric/definition.go

```diff
@@ -1017,11 +1027,46 @@ func isPersistentVolume() definition.FetchFunc {
 	}
 }
 
-func computePercentage(current, all uint64) (definition.FetchedValue, error) {
-	if all == uint64(0) {
-		return nil, errors.New("division by zero")
+func computePercentage(dividend, divisor interface{}) (definition.FetchedValue, error) {
+	var a, b float64
+
+	a, err := convertValue(dividend)
+	if err != nil {
+		return nil, fmt.Errorf("casting dividend: %w", err)
+	}
+
+	b, err = convertValue(divisor)
+	if err != nil {
+		return nil, fmt.Errorf("casting divisor: %w", err)
+	}
+
+	if b == float64(0) {
+		return nil, fmt.Errorf("division by zero")
+	}
+
+	return a / b * 100, nil
+}
+
+func convertValue(v interface{}) (float64, error) {
+	switch v := v.(type) {
+	case uint64:
+		return float64(v), nil
+	case int:
+		return float64(v), nil
+	case prometheus.GaugeValue:
+		return float64(v), nil
+	case float64:
+		return v, nil
+	case definition.FetchedValues:
+		if len(v) == 1 {
+			for _, k := range v {
+				return convertValue(k)
+			}
+		}
+		return 0, fmt.Errorf("unable to convert FetchedValues")
```

Comment:
And get rid of one indentation level by erroring out earlier
```suggestion
		// If retrieved metric is a list of values, we can only handle the case where
		// the list has only one.
		if len(v) != 1 {
			return 0, fmt.Errorf("unable to convert FetchedValues")
		}
		for _, m := range v {
			return convertValue(m)
		}
```

Reply:
makes sense!